tests/flight/sync.rs

```diff
@@ -0,0 +1,280 @@
+use crate::flight::*;
+
+pub(crate) fn sync_cmd_to_flight_data(
+    cmd: DataSyncCommand,
+    batch: RecordBatch,
+) -> FlightDataEncoder {
+    let descriptor = FlightDescriptor::new_cmd(cmd.as_any().encode_to_vec());
+    let flight_stream_builder =
+        FlightDataEncoderBuilder::new().with_flight_descriptor(Some(descriptor));
+    flight_stream_builder.build(futures::stream::iter(vec![Ok(batch)]))
+}
+
+async fn do_put_sync(
+    cmd: DataSyncCommand,
+    batch: RecordBatch,
+    client: &mut FlightClient,
+) -> Result<DataSyncResult> {
+    let flight_data = sync_cmd_to_flight_data(cmd.clone(), batch);
+    let response = client.do_put(flight_data).await?.next().await.unwrap()?;
```

Comment:
Why do we do await().next().await?

Reply:
Because the `do_put` call is itself async (first await), but it also returns a [stream](https://github.com/apache/arrow-rs/blob/c2b05cdbcb37f46f170c7b5073ee6bc2178fdace/arrow-flight/src/sql/server.rs#L382) which we need to await in order to fetch the result. 

In our case we only stream one element back (result) so it''s not very useful, but the mechanism is there for more complex stuff.